MovieApp/PresentationLayer/HomeScreen/HomeScreenViewController.swift

```diff
@@ -98,7 +100,16 @@ class HomeScreenViewController: UIViewController {
         }
 
         popularMoviesCollectionView.onCategoryChanged = { [weak self] optionId in
-            self?.loadPopularMovies(for: optionId)
+            guard let self = self else { return }
+
+            self.presenter
```

Comment:
This is semi-correct :).
Every time a category is selected, you will create a new subscription to fetch new data. You know that only 1 value will be emitted, but the function's signature is AnyPublisher - that means that you really do not know how many values will be emitted.
So, let us analyze this. Category 1 is selected, you will create a new subscription and draw movies. Category 2 is selected, you will create a new subscription and draw movies. While you are on Category 2, a value from Category 1 is emitted and the UI redraws with Category 1 values even though Category 2 is selected.
There are 2 solutions:
1. Add disposables = [] before this line to destroy previous subscriptions
2. Write this in a more reactive way so that popularMoviesCollectionView exposes `var selectedCategoryId: AnyPublisher<Int, Never>` and you use flatMap to load movies and apply it to UI:
```
popularMoviesCollectionView
    .selectedCategoryId
    .flatMap { [weak self] categoryId in
        self?.presenter.popularMovies(for: categoryId) ?? .never()
    }
    .sink { ... }
    .store(in: &disposables)
```

For `selectedCategoryId` you can use PasstroughSubject in `popularMoviesCollectionView`.